src/elements/checkbox/checkbox.js

```diff
@@ -0,0 +1,59 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.checkbox', [])
+
+    .directive('smCheckbox', smCheckbox);
+
+  function smCheckbox() {
+    return {
+      restrict: 'E',
+      require: '?ngModel',
+      transclude: true,
+      scope: {
+        ngDisabled: '='
+      },
+      template: '<div class="ui checkbox" ng-class="{ checked: isChecked }">' +
```

Comment:
for performance, I think we should avoid using `ng-class` for this. An isolate scope is probably a bit too heavy, as well. ng-transclude is also a bit heavy for a form control that may be repeated numerous times on a page. Imho, the impact on the scope hierarchy should be minimal, while still allowing the use of some "angular-isms" (declarative api, ng-model, etc).

I think this is a fine first attempt, and I would be okay with landing this and incrementally improving on it. Love to hear @m0t0r's opinion on this before landing